src/schema/city/index.ts

```diff
@@ -83,6 +89,7 @@ const CityType = new GraphQLObjectType<any, ResolverContext>({
           has_location: true,
           at_a_fair: false,
           ...(args.partnerType && { partner_types: args.partnerType }),
+          ...(args.dayThreshold && { day_threshold: args.dayThreshold }),
```

Comment:
Gotta watch out with this if you expect `0` to be a valid value, as JS treats it as falsy and thus you’d try to spread `0` into the object, which I’m sure is invalid—although I have no idea off-hand what would happen, so perhaps test it in a node REPL.

Reply:
Good thinking @alloy.

Couple of thoughts: passing a 0 might be allowed, but would pretty much by definition return an empty result set.

So it would be both permitted and useless. However — I don't think it would even get sent over to G in the first place:

REPL test:

```js
// here is a normal threshold value

> args = {dayThreshold: 42}
{ dayThreshold: 42 }
> gravityArgs = { foo: "bar", ...(args.dayThreshold && { day_threshold: args.dayThreshold }) } 
{ foo: 'bar', day_threshold: 42 }
> 
> 
// here is the questionable 0 threshold value

> args = {dayThreshold: 0}
{ dayThreshold: 0 }
> gravityArgs = { foo: "bar", ...(args.dayThreshold && { day_threshold: args.dayThreshold }) } 
{ foo: 'bar' }
```

Seems safe to me as-is?

Reply:
yah, looks it 👍

Reply:
thanks all! it should also be safe from Gravity side 🙌 thanks for checking

Reply:
Thanks for verifying @anandaroop! 🙌 

![](https://i.imgur.com/B4D7Lf1.png)

Reply:
For completeness I tested the other falsy values I know of:

```
> { foo: "bar", ...("" && { baz: "bla" }) }
{ foo: 'bar' }
> { foo: "bar", ...(Number.NaN && { baz: "bla" }) }
{ foo: 'bar' }
> { foo: "bar", ...(false && { baz: "bla" }) }
{ foo: 'bar' }
> { foo: "bar", ...(null && { baz: "bla" }) }
{ foo: 'bar' }
```

So I guess we can conclude all falsy values give us our preferred behaviour.

Reply:
Numbers appear to as well, not just zero.

Reply:
@ashfurrow Hmm, not sure I follow:

```
> { foo: "bar", ...(42 && { baz: "bla" }) }
{ foo: 'bar', baz: 'bla' }
```

Reply:
Right, since 42 isn’t falsy. But spreading over 42 works: `{...(42)} = {}`, which is kinda neat!

Reply:
Ohhh, good to know! I guess anything non-iterable gets ignored